src/txn/txn_rollback_to_stable.c

```diff
@@ -766,14 +737,23 @@ __rollback_abort_ondisk_kv(WT_SESSION_IMPL *session, WT_REF *ref, WT_ROW *rip, u
         return (0);
     }
 
-    if (rip != NULL)
-        WT_ERR(__rollback_row_modify(session, page, rip, upd));
-    else
+    if (rip != NULL) {
+        if (row_key != NULL)
+            key = row_key;
+        else {
+            /* Unpack a row key. */
+            WT_ERR(__wt_scr_alloc(session, 0, &key));
+            WT_ERR(__wt_row_leaf_key(session, page, rip, key, false));
+        }
+        WT_ERR(__rollback_row_modify(session, ref, upd, key));
+    } else
         WT_ERR(__rollback_col_modify(session, ref, upd, recno));
     return (0);
```

Comment:
The `key` allocated doesn't get freed leads to the scratch buffer leak in the success scenarios. Rearrange the code to free the allocated scratch buffer.

Reply:
Thanks for spotting it, done in a8462afaa7e7b934ef8f7c53ffc169c16be6bde8